core/transport/src/grpc/meta_master.proto

```diff
@@ -115,12 +115,19 @@ message GetMetricsPResponse {
 message SetPathConfigurationPOptions {}
 message SetPathConfigurationPRequest {
   required string path = 1;
-  required string key = 2;
-  required string value = 3;
-  optional SetPathConfigurationPOptions options = 4;
+  map<string, string> properties = 2;
+  optional SetPathConfigurationPOptions options = 3;
 }
 message SetPathConfigurationPResponse {}
 
+message RemovePathConfigurationPOptions {}
+message RemovePathConfigurationPRequest {
```

Comment:
the meaning of this type isn't very clear - does the nth path correspond to the nth key? Or does each key get applied to each path? I'd like to make it as clear as possible from this type signature what the request means

Reply:
there is just one path and a list of keys for that path, what's the confusing part?

Reply:
oh sorry my eyes are bad, thought `path` was `repeated` not `required` >.<

on that note, we have a best practice of never using `required`. There is little benefit, and it can make it harder to evolve the API in the future. Better to use `optional` instead